src/abstractdataframe/io.jl

```diff
@@ -133,8 +135,8 @@ function writetable(filename::AbstractString,
         # When 'append'-ing to a nonempty file,
         # 'header' triggers a check for matching colnames
         if header
-            if any(i -> @compat(Symbol(file_df[1, i])) != index(df)[i], 1:size(df, 2))
-                throw(KeyError("Column names don't match names in file"))
+            if any(i -> @compat(Symbol(file_df[1, i])) != names(df)[i], 1:size(df, 2))
+                throw(KeyError("Column names in dataframe don't match column names from header in file " * filename))
```

Comment:
While you're at it, could you print the name of the header which triggered the error? That's really really useful when trying to understand what's going wrong.